skimage/transform/_orientation.py

```diff
@@ -0,0 +1,265 @@
+import numpy as np
+
+
+def _normalize(x):
+    """Normalizes an array.
+
+    Parameters
+    ----------
+    x : array_like
+        Array to normalize.
+
+    Returns
+    -------
+    u : array
+        Unitary array.
+
+    Examples
+    --------
+    >>> x = np.arange(5)
+    >>> uX = _normalize(x)
+    >>> np.isclose(np.linalg.norm(uX), 1)
+    True
+    """
+    u = np.asarray(x)
+
+    norm = np.linalg.norm(u)
+
+    return u / norm
+
+
+def _axis_0_rotation_matrix(u, indices=None):
+    """Generate a matrix that rotates a vector to coincide with the 0th (y-)
+       coordinate axis.
+
+    Parameters
+    ----------
+    u : (N, ) array
+        Unit vector.
+    indices : sequence of int, optional
+        Indices of the components of `axis` that should be transformed.
+        If `None`, defaults to all of the indices of `axis`.
+
+    Returns
+    -------
+    R : (N, N) array
+        Orthogonal projection matrix.
+
+    References
+    ----------
+    .. [1] Ognyan Ivanov Zhelezov. One Modification which Increases Performance
+           of N-Dimensional Rotation Matrix Generation Algorithm. International
+           Journal of Chemistry, Mathematics, and Physics, Vol. 2 No. 2, 2018:
+           pp. 13-18. https://dx.doi.org/10.22161/ijcmp.2.2.1
+    .. [2] https://en.wikipedia.org/wiki/Rotation_matrix#In_three_dimensions
+
+
+    Examples
+    --------
+    >>> R = _axis_0_rotation_matrix([0, 1, 0])
+    >>> R @ [0, 1, 0]
+    array([ 1.,  0.,  0.])
+    """
+    ndim = len(u)
+
+    if indices is None:
+        indices = range(ndim)
+
+    x = u
+    w = indices
```

Comment:
Maybe it would be better only using `indices` and `u` as the input anyway. I would recommend you giving better names to the following variables: `R`, `A`, `r`, and so on.